### PR TITLE
feat(VanMoof): add presence

### PR DESCRIPTION
**Pull request by 7loud**
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>

image
image
image
image
</details>

### CODE REVIEW

File: websites/V/VanMoof/presence.ts [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+					presenceData.state = "Help";
+
+					break;
+
+				case "profile":
+					presenceData.state = "Profile";
+
+					break;
+
+				default:
```
Comment (EncryptedDev) [inline websites/V/VanMoof/presence.ts:109]: Put something in the default or remove it

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Comment by Timeraa** [comment]
Adds a Presence for the Company bikes

File: websites/V/VanMoof/presence.ts [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+				case "help":
+					presenceData.state = "Help";
+
+					break;
+
+				case "profile":
+					presenceData.state = "Profile";
+
+					break;
```
Comment (Slowlife01) [inline websites/V/VanMoof/presence.ts:101]:
```suggestion
				case "orders":
				case "rewards":
				case "help":
				case "profile":
					presenceData.state = `${urlpath[3][0].toUpperCase()}${urlpath[3].slice(1)}`;
					break;
```

File: websites/V/VanMoof/presence.ts [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+					presenceData.largeImageKey = "vanmoof_v";
+					presenceData.smallImageKey = "logo";
+					presenceData.buttons = [
+						{
+							label: "View on VanMoof",
+							url: "https://www.vanmoof.com/vanmoof-v/"
+						}
+					];
+
+					break;
```
Comment (Slowlife01) [inline websites/V/VanMoof/presence.ts:226]:
```suggestion
				case "s3":
				case "x3":
				case "vanmoof-v":
					presenceData.details = "Viewing E-Bike";
					presenceData.state = `VanMoof ${urlpath[2].split("-").pop().toUpperCase()}`;
					presenceData.largeImageKey = `vanmoof_${urlpath[2].split("-").pop()}`;
					presenceData.smallImageKey = "logo";
					presenceData.buttons = [
						{
							label: "View on VanMoof",
							url: `https://www.vanmoof.com/${urlpath[2]}/`
						}
					];

					break;
```

**Comment by Bas950** [comment]
VanMoof ❤️

File: websites/V/VanMoof/presence.ts [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+							document.querySelectorAll(".m-info__top-title")[1]?.textContent;
+
+						presenceData.state = `Managing Bike${
+							bikeName ? `: ${bikeName}` : ""
+						}`;
+					}
+
+					break;
+
+				case "orders" || "rewards" || "help" || "profile":
```
Comment (Slowlife01) [inline websites/V/VanMoof/presence.ts:91]: This is not how it works

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

File: websites/V/VanMoof/presence.ts [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+				{
+					label: "View Page",
+					url: window.location.href.replace(`${urlpath[1]}/`, "")
+				}
+			]
+		};
+
+	switch (true) {
+		case /[a-z]{2}(-)[A-Z]{2}/g.test(urlpath[1]) && !urlpath[2]:
+			(presenceData.details = "Home"), delete presenceData.buttons;
```
Comment (Bas950) [inline websites/V/VanMoof/presence.ts:22]: please dont do this, just add the {} and give each its own line

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/V/VanMoof/presence.ts (hunk 5 above)
Comment (N0chteil) [reply websites/V/VanMoof/presence.ts:22]: Edited (https://github.com/PreMiD/Presences/pull/5925)

**Pull request closed without merging**